### PR TITLE
chore: Override smithy-client to fix CodeQL warning

### PR DESCRIPTION
**Pull request by dani-chmb**
*Issue #, if available:*

[Code scanning alert](https://github.com/aws-actions/amazon-ecs-render-task-definition/security/code-scanning/1)

*Description of changes:*

The code-scanning alert points towards this [generated line](https://github.com/aws-actions/amazon-ecs-render-task-definition/blob/f0e3859cc4f50aa73ee4a2dc82781f8737bcbffd/dist/index.js#L21553-L21553) as the security issue. The root cause seems to be the `smithy-client` dependency inherited from the `aws-sdk/client-ecs` devDependency declared in `package.json`. Indeed, the latest version of the smithy-client [exports the "quoteHeader" function](https://www.npmjs.com/package/@smithy/smithy-client?activeTab=code) where the problematic line stems from. Path: `@smithy/smithy-client/dist-types/ts3.4/quote-header.d.ts`

Adding the `@smithy/smithy-client": "3.3.6"` override to `package.json`, followed by this string of commands:

```
% rm -rf node_modules
% rm package-lock.json
% npm install
% npm ci
% npm run package
```
Results in `dist/index.js` no longer containing the problematic line. You should see in the diff for this file the *removal* of:

```
// src/quote-header.ts
function quoteHeader(part) {
  if (part.includes(",") || part.includes('"')) {
    part = `"${part.replace(/"/g, '\\"')}"`;
  }
  return part;
}
__name(quoteHeader, "quoteHeader");
```

The changes were tested by [running a workflow](https://github.com/dani-chmb/simple-ecs-application/actions/runs/11980636297/job/33405292190) that calls the action defined by this PR's commit from a [simple ecs application](https://github.com/dani-chmb/simple-ecs-application/blob/main/.github/workflows/deployInECS.yml#L51). 

Screenshot 2024-11-22 at 16 48 10

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "aws-actions-amazon-ecs-render-task-definition",
-  "version": "1.6.0",
+  "version": "1.6.1",
```
Comment (omkhegde) [inline package.json:3]: why are we manually updating our release version?

Comment (kg-aws) [reply package.json:3]: +1

**Review by omkhegde — CHANGES_REQUESTED** [review]
Added comment

File: package.json (hunk 1 above)
Comment (dani-chmb) [reply package.json:3]: Ack, will remove manual update of release version

**Comment by dani-chmb** [comment]
Addressed version number comment with new commit.

**Review by omkhegde — APPROVED** [review]

**Review by karanbokil — APPROVED** [review]

**Pull request merged**